layers/+lang/c-c++/packages.el

```diff
@@ -182,75 +186,45 @@
+      (add-to-list 'projectile-globally-ignored-directories c-c++-lsp-cache-dir)
+      (when c-c++-adopt-subprojects
+        (setq projectile-project-root-files-top-down-recurring
+              (append '("compile_commands.json"
+                        ".cquery"
+                        ".ccls")
+                      projectile-project-root-files-top-down-recurring))))))
+
 (defun c-c++/init-rtags ()
+  ;; config in `funcs.el'
   (use-package rtags
-    :if (eq c-c++-backend 'rtags)
-    :init
-    (progn
-      (setq rtags-autostart-diagnostics t)
-      (add-hook 'rtags-jump-hook 'evil-set-jump)
-      (rtags-diagnostics)
-      ;; key bindings
-      (evil-define-key 'normal rtags-mode-map
-        (kbd "RET")   'rtags-select-other-window
-        (kbd "M-RET") 'rtags-select
-        (kbd "q")     'rtags-bury-or-delete)
-      ;; TODO check for consistency with gtags key bindings
-      ;; see https://github.com/syl20bnr/spacemacs/blob/develop/layers/+tags/gtags/funcs.el#L70
-      (dolist (mode c-c++-modes)
-        (spacemacs/set-leader-keys-for-major-mode mode
-          "g." 'spacemacs/c-c++-tags-find-symbol-at-point
-          "g," 'spacemacs/c-c++-tags-find-references-at-point
-          "g;" 'spacemacs/c-c++-tags-find-file
-          "g/" 'rtags-find-all-references-at-point
-          "g[" 'rtags-location-stack-back
-          "g]" 'rtags-location-stack-forward
-          "g>" 'spacemacs/c-c++-tags-find-symbol
-          "g<" 'spacemacs/c-c++-tags-find-references
-          "gB" 'rtags-show-rtags-buffer
-          "gd" 'rtags-print-dependencies
-          "gD" 'rtags-diagnostics
-          "ge" 'rtags-reparse-file
-          "gE" 'rtags-preprocess-file
-          "gF" 'rtags-fixit
-          "gG" 'rtags-guess-function-at-point
-          "gh" 'rtags-print-class-hierarchy
-          "gI" 'spacemacs/c-c++-tags-imenu
-          "gL" 'rtags-copy-and-print-current-location
-          "gM" 'rtags-symbol-info
-          "gO" 'rtags-goto-offset
-          "gp" 'rtags-set-current-project
-          "gR" 'rtags-rename-symbol
-          "gs" 'rtags-print-source-arguments
-          "gS" 'rtags-display-summary
-          "gT" 'rtags-taglist
-          "gv" 'rtags-find-virtuals-at-point
-          "gV" 'rtags-print-enum-value-at-point
-          "gX" 'rtags-fix-fixit-at-point
-          "gY" 'rtags-cycle-through-diagnostics)))))
+    :defer t))
 
 (defun c-c++/post-init-realgud()
   (dolist (mode c-c++-modes)
     (spacemacs/add-realgud-debugger mode "gdb")))
 
 (defun c-c++/post-init-semantic ()
-  (spacemacs/add-to-hooks 'semantic-mode c-c++-mode-hooks)
-  (when (or (configuration-layer/layer-used-p 'gtags)
-            (configuration-layer/layer-used-p 'lsp))
-    (spacemacs/add-to-hooks 'spacemacs//disable-semantic-idle-summary-mode c-c++-mode-hooks t)))
+  (add-hook 'c-mode-local-vars-hook #'spacemacs//c-c++-setup-semantic)
+  (add-hook 'c++-mode-local-vars-hook #'spacemacs//c-c++-setup-semantic))
```

Comment:
I encountered an error here saying spacemacs//c-c++-setup-semantic is a void-function. I can't find this function defined anywhere, am I missing something?

Reply:
Thank you for catching this, I added semantic config at the end and forgot to add the dispatch function.
Should be fixed.